### PR TITLE
Split `CHANGELOG.md` files for beta and stable builds

### PR DESCRIPTION
**Pull request by prprabhu-ms**
Alternative for #2504 

Split the current `CHANGELOG`s for `stable` vs `beta` channel releases. A follow up PR will update the tooling so that future releases correctly update the relevant `CHANGELOG`s.

- Split the change files per build flavor.
- Split `CHANGELOG.md` files for `@azure/communication-react` packlet into two flavors
- Delete `CHANGELOG.md` files for all other packlets. *They were never groomed manually, and nobody ever looked at them*.
- Update `.gitignore` so we don't accidentally commit a build flavor-unaware `CHANGELOG`.
- Add a top-level `CHANGELOG.md` that points to the correct file deep in our tree. The lack of this file has been very confusing for devs, even on our own team.

Note that the `CHANGELOG.json` files are _not_ split as the JSON data is already keyed by the release version.

### CODE REVIEW

File: change/@internal-calling-stateful-client-8010c09c-1bc6-4f82-9a12-7d2383b46324.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "boo",
```
Comment (prprabhu-ms) [inline change/@internal-calling-stateful-client-8010c09c-1bc6-4f82-9a12-7d2383b46324.json:3]: Err... did I _really_ mean to add these?

File: packages/acs-ui-common/CHANGELOG.md [hunk 2]
```diff
@@ -1,94 +0,0 @@
-# Change Log - @internal/acs-ui-common
```
Comment (JamesBurnside) [inline packages/acs-ui-common/CHANGELOG.md:1]: Should the `changelog.json` also be deleted per packlet? (and just have one under `communication-react`)

File: change-beta/@internal-acs-ui-common-358e1672-866d-4320-b7c4-48b8765054c4.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "something",
```
Comment (JamesBurnside) [inline change-beta/@internal-acs-ui-common-358e1672-866d-4320-b7c4-48b8765054c4.json:3]: assume these test ones are being deleted

File: packages/acs-ui-common/CHANGELOG.md (hunk 2 above)
Comment (prprabhu-ms) [reply packages/acs-ui-common/CHANGELOG.md:1]: Can't quite delete it right now. As long as there are changefiles for the packlets, collecting the changelogs will update these JSON files.

Once we're done with the existing changefiles for these packlets (i.e., once these PRs land and we've released a beta and a stable release), we can consider deleting the various CHANLOG.json files.

**Review by JamesBurnside — APPROVED** [review]

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**